admintools/middleware.py

```diff
@@ -29,31 +33,30 @@ def process_request(self, request):
                     rollbar.report_message('Impossible case: attempting to emulate another user but not specified')
 
                 if targeted_user is None:
-                    # TODO: message - 'User does not exist'
+                    messages.error(request, 'Cannot Emulate: User does not exist')
                     pass
                 else:
                     if is_allowed_to_emulate(request.user, targeted_user):
                         request.original_user = request.user
                         request.user = targeted_user
-                        # TODO: add message indicating successfully emulated
                     else:
-                        # TODO: add message indicating failure to emulate
+                        messages.error(request, 'Cannot Emulate: Not allowed to emulate that user')
                         pass
             else:
                 # not attempting to emulate
                 pass
         else:
-            # the original user is not allowed to emulate
+            messages.error(request, 'Cannot Emulate: Not allowed to emulate that user')
             pass
 
+
     def process_response(self, request, response):
         """Delete user emulation cookies if they should not be set
         """
         original_user = getattr(request, 'original_user', None)
         user = getattr(request, 'user', None)
         is_emulating = original_user is not None and user is not None
-
-        if not(is_emulating or is_allowed_to_emulate_users(user)):
+        if not is_emulating:
```

Comment:
I changed this back, we need to have a final decision on it, but it was annoying for me to be able to not be able to emulate as a non-existant user or as a super user but appear to be emulating as them in the htk tab and toolbar.

@jontsai I simplified the logic. Since our process_request is doing the grunt of the work, we just need to simply look at the original_user to decide if we want to delete cookies.

Reply:
that seems fine

Reply:
yeah, it makes sense, because you don't want that alert to keep popping up on every page load, I suppose